.github/workflows/action_tests.yaml

```diff
@@ -202,3 +202,64 @@ jobs:
             echo "The did not have the expected latest commit message "$latest_commit_message"."
             exit 1
           fi
+
+  payload:
+    name: ${{ matrix.description }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - description: pull-request-payload
+            payload_path: action_tests/pull_request_test_payload.json
+          - description: trunk-merge-payload
+            payload_path: action_tests/trunk_merge_test_payload.json
+          - description: all-payload
+            payload_path: action_tests/all_test_payload.json
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          path: local-action
+
+      - name: Checkout
+        shell: bash
+        run: |
+          mv local-action ..
+          ../local-action/action_tests/setup.sh /tmp/src-repo .
+          mv ../local-action .
+
+      - name: Craft TEST_GITHUB_EVENT_PATH
+        shell: bash
+        run: |
+          payload=$(jq --null-input --argfile foo local-action/${{ matrix.payload_path }} '{"payload": ($foo | tostring)}')
+          input=$(echo '{"inputs":'$(echo $payload | jq -c '.' )'}')
+          echo "$input">>local-action/payload.json
```

Comment:
nit: I think this can be combined into, although mixing the quotes might make things hard to read
```sh
echo '{"inputs":"$(echo $payload | jq -c '.' )"}'>>local-action/payload.json
```